pandas/core/ops.py

```diff
@@ -1003,8 +1003,18 @@ def _arith_method_SERIES(cls, op, special):
                         if op is divmod else _construct_result)
 
     def na_op(x, y):
-        import pandas.core.computation.expressions as expressions
+        # handle extension array ops
+        # TODO(extension)
+        # the ops *between* non-same-type extension arrays are not
+        # very well defined
+        if (is_extension_array_dtype(x) or is_extension_array_dtype(y)):
+            if (op_name.startswith('__r') and not
+                    is_extension_array_dtype(y) and not
+                    is_scalar(y)):
+                y = x.__class__._from_sequence(y)
+            return op(x, y)
```

Comment:
Is it really necessary to put this in the closure?  It seems like this case could be handled directly on line 1061.

Reply:
the way the r ops are defined is not compatible with some types, pretty much if you have a 

``rop(et, ndarray)`` will fail for some types, e.g. say ndarray is a datetime64[ns] then et is an integer like the precedence is wrong (IOW need to defer to the op here)

Reply:
I think I understand what you're saying, but that it isn't what I'm getting at.  Let me try to rephrase:

In `wrapper` below, there is currently (in the PR)

```
        elif (is_extension_array_dtype(left) or
                is_extension_array_dtype(right)):
            pass
```

I'm suggesting that instead of `pass` there, the code above (currently 1006-1015) could go there.  The main reason why this might _not_ be appropriate is if the try/except in `safe_na_op` is relevant for this case.